modules/bluefish/bluefish.cpp

```diff
@@ -78,21 +75,42 @@ std::vector<std::wstring> device_list()
 
 void init(core::module_dependencies dependencies)
 {
+
+	bool bCanBeConsumer = false;
 	try
 	{
-		blue_initialize();
+		BvcWrapper blue = BvcWrapper();
+		int numCards = 0;
+		blue.Enumerate(numCards);
+		if (numCards)
+		{
+			for (int n = 1; n < (numCards + 1); n++)
+			{
+				unsigned int value;
+				blue.Attach(n);
+				blue.QueryCardProperty32(CARD_FEATURE_STREAM_INFO, value);
+				if (CARD_FEATURE_GET_SDI_OUTPUT_STREAM_COUNT(value))
+					bCanBeConsumer = true;
+
+				blue.Detach();
+			}
+		}	
 	}
 	catch(...){}
 
-	dependencies.consumer_registry->register_consumer_factory(L"Bluefish Consumer", create_consumer, describe_consumer);
-	dependencies.consumer_registry->register_preconfigured_consumer_factory(L"bluefish", create_preconfigured_consumer);
-	dependencies.system_info_provider_repo->register_system_info_provider([](boost::property_tree::wptree& info)
+	if (bCanBeConsumer)
```

Comment:
It is better if the consumer is registered even though there are no capable Bluefish cards in the computer, so for example

```
HELP CONSUMERS
```

...always includes Bluefish Consumer even though it is not possible to instantiate one based on the current system configuration.